code/modules/spells/spell_types/blind.dm

```diff
@@ -1,34 +1,47 @@
-/obj/effect/proc_holder/spell/pointed/trigger/blind
+/obj/effect/proc_holder/spell/targeted/blind
 	name = "Blind"
 	desc = "This spell temporarily blinds a single target."
 	school = "transmutation"
 	charge_max = 300
 	clothes_req = FALSE
 	invocation = "STI KALY"
 	invocation_type = "whisper"
-	message = "<span class='notice'>Your eyes cry out in pain!</span>"
 	cooldown_min = 50 //12 deciseconds reduction per rank
-	starting_spells = list("/obj/effect/proc_holder/spell/targeted/inflict_handler/blind", "/obj/effect/proc_holder/spell/targeted/genetic/blind")
 	ranged_mousepointer = 'icons/effects/blind_target.dmi'
 	action_icon_state = "blind"
-	base_icon_state = "blind"
-	active_msg = "You prepare to blind a target..."
+	range = 7
+	selection_type = "range"
+	var/duration = 300 //30 seconds
+	var/static/list/compatible_mobs_typecache = typecacheof(list(/mob/living/carbon/human))
 
-/obj/effect/proc_holder/spell/targeted/inflict_handler/blind
-	amt_eye_blind = 10
-	amt_eye_blurry = 20
-	sound = 'sound/magic/blind.ogg'
 
-/obj/effect/proc_holder/spell/targeted/genetic/blind
-	mutations = list(BLINDMUT)
-	duration = 300
-	charge_max = 400 // needs to be higher than the duration or it'll be permanent
-	sound = 'sound/magic/blind.ogg'
+/obj/effect/proc_holder/spell/targeted/blind/cast(list/targets, mob/user = usr)
+	if(!targets.len)
```

Comment:
`length()` is more performant than `.len` if you're just checking list length :)
```suggestion
	if(!length(targets))
```